packages/utilities/src/GlobalSettings.ts

```diff
@@ -61,28 +46,32 @@ export interface IChangeEventCallback {
  */
 export class GlobalSettings {
   public static getValue<T>(key: string, defaultValue?: T | (() => T)): T {
-    if (_globalSettings[key] === undefined) {
-      _globalSettings[key] = typeof defaultValue === 'function' ? defaultValue() : defaultValue;
+    const globalSettings = _getGlobalSettings();
+
+    if (globalSettings[key] === undefined) {
+      globalSettings[key] = typeof defaultValue === 'function' ? (defaultValue as Function)() : defaultValue;
```

Comment:
Not sure why you need the `(defaultValue as Function)` cast since you just checked that it's a function?

Reply:
typescript is not that smart.